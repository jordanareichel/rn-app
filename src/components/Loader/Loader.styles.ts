import {Text} from '@components/Text';
import styled from 'styled-components/native';

export const Wrapper = styled.View`
  flex: 1;
  justify-content: center;
  align-self: center;
`;

export const Title = styled(Text)`
  margin: 16px 0 0;
`;
