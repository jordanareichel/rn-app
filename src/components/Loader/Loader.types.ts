import {ModalProps} from 'react-native';

export type LoaderProps = {
  size?: number | 'small' | 'large';
  text?: string;
  color?: string;
} & ModalProps;
