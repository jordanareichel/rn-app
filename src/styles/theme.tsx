export const theme = {
  font: {
    sm: '12',
    md: '14',
    lg: '18',
  },
  colors: {
    dark: '#000000',
    light: '#ffffff',
  },
};
