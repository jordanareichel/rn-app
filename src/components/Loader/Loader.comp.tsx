import React from 'react';
import {ActivityIndicator, Modal} from 'react-native';
import {Title, Wrapper} from './Loader.styles';
import {LoaderProps} from './Loader.types';

export const Loader: React.FC<LoaderProps> = props => {
  const {size, text, color, ...rest} = props;

  return (
    <Modal {...rest}>
      <Wrapper>
        <ActivityIndicator size={size} color={color} />
        <Title fontWeight="semi-bold">{text}</Title>
      </Wrapper>
    </Modal>
  );
};

Loader.defaultProps = {
  animationType: 'fade',
  color: 'green',
};
