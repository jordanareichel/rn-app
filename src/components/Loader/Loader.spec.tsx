import React from 'react';

import TestRenderer from 'react-test-renderer';
import {ThemeProvider} from 'styled-components/native';
import {Loader} from './Loader.comp';
import {theme} from '@styles/theme';

describe('Component Loader', () => {
  test('renderer currently', () => {
    const tree = TestRenderer.create(
      <ThemeProvider theme={theme}>
        <Loader />
      </ThemeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
