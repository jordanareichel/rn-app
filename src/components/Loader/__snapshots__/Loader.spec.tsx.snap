// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Loader renderer currently 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <ActivityIndicator
      color="green"
    />
    <Text
      color="black"
      fontSize={18}
      fontWeight="semi-bold"
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 18,
            "fontWeight": "semi-bold",
            "textAlign": "center",
          },
          Object {
            "marginBottom": 0,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 16,
          },
        ]
      }
    />
  </View>
</Modal>
`;
